File: problem/digit-count/6/main.js
// main.js: main file for solutions to the digit-count exercise
//
// It processes values from standard input then from the "in.txt" file.
// This is appended to the submitted JavaScript program and tests the digit-count function.
//
// The submitted file processes the standard input and this processes the
// "in.txt" file.
//
// Copyright (C) 2020-2021  Rudy Matela
// All rights reserved.

var fs = require('fs').createReadStream('in.txt');
var rl = require('readline').createInterface({input: fs, output: process.stdout});

rl.on('line', function(line){
  words = line.split(/ /);
  d = parseInt(words[0]);
  x = parseInt(words[1]);
  console.log(digit_count(d,x));
});

File: problem/digit-count/6/main.cc
/*
 * main.c: main file for a solution to the digit-count exercise
 *
 * This is to be linked to the submitted file.
 * It processes values from standard input then from the "in.txt" file.
 *
 *
 * Copyright (C) 2020-2021  Rudy Matela
 * All rights reserved.
 */
#include <stdio.h>
int digit_count(int, int);
int main()
{
	int i, d;
	FILE *f;
	/* process standard input */
	while (scanf(" %d %d", &d, &i)==2)
		printf("%d\n", digit_count(d, i));
	/* process additional inputs from in.txt */
	f = fopen("in.txt", "r");
	if (!f) {
		fprintf(stderr, "could not open in.txt");
		return 1;
	}
	while (fscanf(f," %d %d", &d, &i)==2)
		printf("%d\n", digit_count(d, i));
	fclose(f);
	/* in Python, we cannot load a library without "running" it.
	 * so for this test case, we provide additional inputs on "in.txt"
	 * cf. main.py
	 */
	return 0;
}
